src/renderers/webgl/WebGLProgram.js

```diff
@@ -673,27 +673,29 @@ function WebGLProgram( renderer, cacheKey, parameters, bindingStates ) {
 
 		versionString = '#version 300 es\n';
 
-		prefixVertex = [
-			'#define attribute in',
-			'#define varying out',
-			'#define texture2D texture'
-		].join( '\n' ) + '\n' + prefixVertex;
-
-		prefixFragment = [
-			'#define varying in',
-			( parameters.glslVersion === GLSL3 ) ? '' : 'out highp vec4 pc_fragColor;',
-			( parameters.glslVersion === GLSL3 ) ? '' : '#define gl_FragColor pc_fragColor',
-			'#define gl_FragDepthEXT gl_FragDepth',
-			'#define texture2D texture',
-			'#define textureCube texture',
-			'#define texture2DProj textureProj',
-			'#define texture2DLodEXT textureLod',
-			'#define texture2DProjLodEXT textureProjLod',
-			'#define textureCubeLodEXT textureLod',
-			'#define texture2DGradEXT textureGrad',
-			'#define texture2DProjGradEXT textureProjGrad',
-			'#define textureCubeGradEXT textureGrad'
-		].join( '\n' ) + '\n' + prefixFragment;
+		prefixVertex = /* glsl */`
+			#define attribute in
+			#define varying out
+			#define texture2D texture
+			${prefixVertex}
+		`;
+
+		prefixFragment = /* glsl */`
+			#define varying in
+			${( parameters.glslVersion === GLSL3 ) ? '' : 'out highp vec4 pc_fragColor;'}
+			${( parameters.glslVersion === GLSL3 ) ? '' : '#define gl_FragColor pc_fragColor'}
```

Comment:
Can you please verify if this statement actually works? If this code breaks, it is not visible with a broken E2E test.

Reply:
Sorry, I don't how to verify it, you mean `npm run test-e2e`?

Reply:
You need a special test case for this. Fortunately, I still have a fiddle around: https://jsfiddle.net/rnbqyx25/

The idea is to use a `ShaderMaterial` with GLSL 3.0 code and check if `out highp vec4 pc_fragColor;` and `#define gl_FragColor pc_fragColor` is NOT set.